src/AutorestSwift/Models/Complex/Value.swift

```diff
@@ -87,4 +87,67 @@ class Value: Codable, LanguageShortcut {
         case `protocol`
         case extensions
     }
+
+    // MARK: Codable
+
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let schemaContainer = try container.nestedContainer(keyedBy: Schema.CodingKeys.self, forKey: .schema)
+        let type = try? schemaContainer.decode(AllSchemaTypes.self, forKey: Schema.CodingKeys.type)
+
+        var schema: Schema?
+        switch type {
+        case .array:
+            schema = try? container.decode(ArraySchema.self, forKey: .schema)
+        case .dictionary:
+            schema = try? container.decode(DictionarySchema.self, forKey: .schema)
+        case .object:
+            schema = try? container.decode(ObjectSchema.self, forKey: .schema)
+        case .number:
+            schema = try? container.decode(NumberSchema.self, forKey: .schema)
+        case .choice:
+            schema = try? container.decode(ChoiceSchema.self, forKey: .schema)
+        case .dateTime:
+            schema = try? container.decode(DateTimeSchema.self, forKey: .schema)
+        case .string:
+            schema = try? container.decode(StringSchema.self, forKey: .schema)
+        default:
+            schema = try container.decode(Schema.self, forKey: .schema)
+        }
```

Comment:
This is why an initializer would be nice. Consolidate all of _this_

Reply:
Found a solution to be able to consolidate all Schema creation.